sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/util/SelectStmtUtil.kt

```diff
@@ -15,13 +16,16 @@ import com.intellij.psi.PsiElement
 
 internal sealed class TableNameElement {
```

Comment:
What is the reason for this class? Instead, we could use `NamedElement`...

Reply:
i think it was just to be more explicit about the type being passed around, but having less types in general is probably better, so if there's a way to just delete it entirely I'm not opposed.

Reply:
After refactoring the changed code in `pureTable` I don't use `referencedTable` and `NamedElement` anymore. Do you want these unrelated changes or should I drop them?

Reply:
What does it mean to upvote? :D